src/gui/owncloudgui.cpp

```diff
@@ -173,11 +173,11 @@ void ownCloudGui::slotTrayClicked(QSystemTrayIcon::ActivationReason reason)
     if (reason == QSystemTrayIcon::Trigger) {
         if (OwncloudSetupWizard::bringWizardToFrontIfVisible()) {
             // brought wizard to front
-        } else if (_shareDialogs.size() > 0) {
-            // Share dialog(s) be hidden by other apps, bring them back
-            Q_FOREACH (const QPointer<ShareDialog> &shareDialog, _shareDialogs) {
-                Q_ASSERT(shareDialog.data());
-                raiseDialog(shareDialog);
+        } else if (!_visibleDialogs.isEmpty()) {
+            // Dialog(s) be hidden by other apps, bring them back (e.g. Share / Settings / Auth dialogs)
+            for (const QPointer<QDialog> &dialog : qAsConst(_visibleDialogs)) {
+                Q_ASSERT(dialog.data());
```

Comment:
Well, you got a bunch which are delete on close (including the settings dialog in another PR) so it needs to be ready to handle nullptr here. I'd turn that into:
```
if (!dialog) {
    continue;
}
```

Reply:
done